Assets/MRTK/Core/Definitions/Devices/ArticulatedHandDefinition.cs

```diff
@@ -264,7 +303,7 @@ public void UpdateCurrentIndexPose(MixedRealityInteractionMapping interactionMap
         {
             using (UpdateCurrentIndexPosePerfMarker.Auto())
             {
-                if (unityJointPoses.TryGetValue(TrackedHandJoint.IndexTip, out currentIndexPose))
+                if (unityJointPoses != null)
```

Comment:
Is it the case that if we can't get one joint, we can't get any of them? I think these cases were originally supposed to just check if specific joints were unavailable, while this change checks if any were reported at all.

Reply:
That's true...this does add an "all or nothing" assumption to the internals.
I have another approach around defining a new `IDictionary` type that's backed by an array, and then maybe we return `pose != default(MixedRealityPose)` or something. I can go down that path quickly and see the perf comparisons.

Reply:
This is the last thing that caught my eye in this PR. I think if we can avoid doing the all or nothing that'd be preferrable, though I'm willing to let the behavior fall through here if it's a small edge case and we get lots of perf improvement